fix(Group, ActiveSelection): regression, wrong bounding box position when activeSelection or Group has originX/originY that are not default left/top (#9649)
fabricjs/fabric.js: src/LayoutManager/README.md

Layout manager exposes a main public method that performs group layout given a target group and target objects.

## Expectations for non interactive groups

The layout manager will calculate the size and position of the group when is created or updated
The FitContent strategy is the classic fabric 5.x group and will grow or shrink when objects are added or removed.

> In general the expectation is that creating or modifying an object will create a group with a postion and once that group is added to canvas, the position of the objects will be the same visually that those object would have had on the canvas if they weren't on the group.

Reviewer: This is correct only if left and top are not passed
Could it be you missed that?

Author: is written below indeed in the paragraph about position.

Reviewer: confusing

This is a legacy behaviour that carries over this new code.
The behaviour is not important if you are building a group from scratch and you are adding it in a specific position on the canvas, in that case the original position is meaningless and you are overriding it by specifying your postion.

> Since inside the group all objects are organized around the center is anyway easier if you can think of your objects position and alignment with scene coordinates and know that they will be maintained in the same position when grouped.

Reviewer: ```suggestion
Since inside the group all objects are organized around the center is anyway easier if you can think of your objects position and alignment with scene coordinates and know that they will be maintained in the same position when grouped only if you did not pass transform options.
```

Reviewer: This is very misleading

### FitContent strategy

When a group is created, FitContent strategy will consider all the objects in the group, find their bounding box, find the center of the bounding box, and assign those to the group.

> When a new object is added or removed, a new bounding box is calculated, a new center is found, all the remaining objects are moved by an offset so that are re-centered around the new center

Reviewer: ```suggestion
When a new object is added or removed, a new bounding box is calculated, a new center is found, all the remaining objects are moved by an offset so that are re-positioned around the new center
```

### Fixed layout strategy

With the fixed layout a Developer can specify either width or height and have the other calculated.
If the developer specifies both dimensions of the box, the bounding box is still calculated to obtain the center.

> Once the group is initialized with a fixed layout, adding new objects is not changing the center anymore.

Reviewer: ```suggestion
Once the group is initialized with a fixed layout, adding new objects will not change the group's size and therefore also its position.
```

> The layout is fixed, is cropped by the bounding box itself.

Reviewer: partially correct, cropping is not precise from what I recall

> When larger or smaller than the actual boundingbox the group is centered on the initial center of the initial set of objects and the fixed bounding box extends from the center.

Reviewer: wrong
it used to position by the passed origin and that is what you missed

Author: that is not a feature. we don't do that.
originX and originY represent the anchor point of the object that is located at top/left. They don't represent anything else. They can't also be used to say that since a bounding box is smaller the bbox left/top corner will be aligned wit the top/left corner of the elements, or in another case bottom/right. That is not originX/originY.

Reviewer: I wrote group layout to fix v5 behavior not to organize it
Centering it makes no sense and again this is why this PR should be blocked until you remove originX/Y
If you do not agree with me go ahead
I don't approve this direction.

Author: Well you didn't comunicate that clearly when you organize the group rewrite, at all.
Please try to explain what issue the centering causes apart solving the issue of shifting objects.

### Clippath layout strategy

The Clippath layout keeps the group around the clipath rather than the objects. The group will be as large as the clipPath, the center of the group will be the center of the clippath.
This layout is clippath dependentent

WARNING needs fix:
The behaviour of this layout is currently broken if we want.
The center of the group is the center of the clipPath visually but the position of the clipPath compared to the object on the canvas is hard to predict as it is now

> 

Author: @ShaMan123  I verified those behaviors in master and in the PR.

The differences are that now originX and originY do not count anymore for group layouting.
So for example in master with the fixed strategy the final group bounding box would be aligned with the top,left  corner or the center or the bottom,right corner depending of the group origin.
Object displacement would happen in any case apart the topleft.

The clipPath has a similar behaviour in master with the clipPath ending not in the center if is not using top,left and object being displaced.

Now the final viewport of the group is always aligned with the center of the initial bounding box, but never displaces anything.
The Clippath has an added behaviour that i do not understand, in particular that if you specify a position for the clippath is not clear how that relates with the objects. but this is out of scope for me to fix now, i m fine to know is known bug and move to other things

Reviewer: What you describe is correct and proves to me that this PR is wrong
And that is why I want to block it until you remove originX/Y
